prefect_census/client.py

```diff
@@ -0,0 +1,93 @@
+"""Module containing client for interacting with the Census API"""
+from httpx import AsyncClient, Response
+
+
+class CensusClient:
+    """
+    Client for interacting with the Census API.
+
+    Args:
+        api_key: API key to authenticate with the Census API.
+    """
+
+    def __init__(self, api_key: str):
+        self._closed = False
+        self._started = False
+
+        self.client = AsyncClient(
+            base_url="https://app.getcensus.com",
+            headers={"Authorization": f"Bearer {api_key}"},
```

Comment:
The Census API uses a different method for authentication which is documented here: https://docs.getcensus.com/basics/api#authenticating-api-requests. I think you can use httpx's `auth` parameter for this, but we'll want to test against a real Census instance to verify.